packages/@lwc/ssr-compiler/src/__tests__/utils/expected-failures.ts

```diff
@@ -34,6 +29,7 @@ export const expectedFailures = new Set([
     'scoped-slots/mixed-with-light-dom-slots-inside/index.js',
     'scoped-slots/mixed-with-light-dom-slots-outside/index.js',
     'slot-forwarding/scoped-slots/index.js',
+    'slot-not-at-top-level/advanced/ifTrue/light/index.js',
```

Comment:
@wjhsf @jhefferman-sfdc I ended up having to do some big refactoring to avoid shared state in text node concatenation: f7a1e07f4

This led to a new test failure here because the `cxt.nextSibling` is actually wrong. This is something unrelated to this PR that we should fix independently. I can open a separate issue for that.

Reply:
Actually it looks like only adjacent text concatenation relies on `cxt.nextSibling`. 😆

Reply:
Wrote a minimal repro... let's deal with it later: 7161ea418

Reply:
Fixed that, wrote a better minimal repro... time to stop for real now 😛 34c5e04bb